### ISSUE TITLE
[Side-quest] Implement syntax highlighting PrismCode component

### ISSUE THREAD

**Issue by jeanlescure** [KunePay/kune-ui#15]
# Side Quest

![](https://thumbs.gfycat.com/YellowEcstaticCoyote-size_restricted.gif)

Things are starting to take form, we need to have syntax highlighting for the code boxes under each component documented under the *Components* page.

The ideal solution should be react version agnostic to be able to maintain long term support. MVP should be fully compatible with the current version we're using.

**Comment by jeanlescure** [comment]
Most serious react implementations use a version of [PrismJS](https://prismjs.com).

The package [react-prism](https://github.com/tomchentw/react-prism/) seems to be pretty stable.

Documentation for both is lacking detail, but implementation seems to be simple enough that I should be able to circumvent any unknowns.

**Comment by jeanlescure** [comment]
The `PrismCode` component is straight-forward enough that it would be a wasted effort to abstract further.

Changing the scope of this issue to just implement `PrismCode` in the *Components* page.

**Comment by jeanlescure** [comment]
The following is all that's needed to use `PrismCode`  in a Next.js page:

```js
require('prismjs');
require('prismjs/components/prism-jsx'); // Without this jsx highlighting doesn't work
require('prismjs/themes/prism-okaidia.css');

import {PrismCode} from 'react-prism';
```

And to use is as simple as:

```jsx
<PrismCode component="pre" className="language-jsx">
{`
<Heading level={4} className="highlight">.highlight</Heading>
`}
</PrismCode>
```

It should be noted that in order for this line to work:

```js
require('prismjs/themes/prism-okaidia.css');
```

I had to update to the latest version of Next.js and Babel. Doing this upgrade only affected one HOC (`withReduxStore`). A bit annoying since the error was super ambiguous and undocumented. But it was fixed on [abd197ba70fc1bcac0933a83e4a236b114f4539d](https://github.com/KunePay/kune-ui/commit/abd197ba70fc1bcac0933a83e4a236b114f4539d).
